lms/services/digest/_models.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from functools import cached_property
+from typing import List
+
+
+@dataclass
+class HUser:
+    """A class representing H users."""
+
+    h_userid: str
+    """The H userid like 'acct:name@lms.example.com'."""
+
+    display_name: str
+    """A human readable display name."""
+
+    email: str
+    """Email for the user."""
+
+
+@dataclass
+class HCourse:
+    """A class representing courses as groups in H."""
+
+    title: str
+    """Title for the course."""
+
+    authority_provided_id: str
+    """The course's authority provided id."""
+
+    aka: List[str]
+    """All authority provided ids which map to the course from groups etc."""
+
+    instructors: List[str]
+    """A list of H userids for users who are instructors for the course."""
+
+    annotations: List[dict] = None
+    """A list of annotations in the course."""
```

Comment:
This PR would be more compact with less doc-strings, but we don't actually get any prizes for few lines if they are there for visual clarity.

I think spacing things out enough can eventually become a readability issue, but this makes it easier to associate the comment to the attr I think. If you have them all squished, or not documented, it's a bit less clear what things are.

We can try to get around that with naming, but `list_of_instructor_h_userids` or `authority_provided_id_for_self_and_children` etc. kind of suck.